internal/config/config.go

```diff
@@ -0,0 +1,59 @@
+// Package config provides functionality for loading configuration
+// from a YAML file and environment variables.
+package config
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Config describes the structure of the server and database configuration.
+type Config struct {
+	Server struct {
+		Port int `yaml:"port"`
+	} `yaml:"server"`
+
+	PostgreSQL struct {
+		Host          string `yaml:"host"`
+		Port          int    `yaml:"port"`
+		Authorisation struct {
+			Env struct {
+				LoginEnv    string `yaml:"login"`
+				PasswordEnv string `yaml:"password"`
+			} `yaml:"env"`
+		} `yaml:"authorisation"`
+	} `yaml:"postgresql"`
+
+	DBLogin    string
```

Comment:
You don't fill it in cfg, yet you use it to craft a dsn
You should read environment variables to obtain login and password